### DISCUSSION TITLE
Why is my test accuracy so low even though Daniel's so so high?

### DISCUSSION THREAD

**Discussion by GodRishUniverse** [mrdbourke/pytorch-deep-learning#861] [Q&A] [answered]
This is somewhere between chapter 79 and 84 of his course. My accuracy is:
![image](https://github.com/mrdbourke/pytorch-deep-learning/assets/69109851/c1812708-5a6e-4aaa-b33a-f77398ada1ae)
While Daniel's is:
![image](https://github.com/mrdbourke/pytorch-deep-learning/assets/69109851/4b78dce2-f63b-4356-9fd6-7f2abf3c1394)

Here is my code for accuracy: 
```# Calculate accuracy - out of a 100 examples - what percentage does our model get it right

def accuracy_fn(y_true,y_pred):
  # torch.eq tells how many samples in y_true and y_preds are equal - sums gets the total and item gets it as a single value
  correct = torch.eq(y_true,y_pred).sum().item()
  acc = (correct/(len(y_preds)))*100
  return acc
```

And here is my code for the Model: 
```#Writing the training loop 

torch.manual_seed(42)
torch.cuda.manual_seed(42)

X_train, y_train, X_test, y_test = X_train.to(device), y_train.to(device), X_test.to(device), y_test.to(device)

epochs =1000

for epoch in range(epoch):
  # Training
  model_3.train()

  #Forward pass
  y_logits= model_3(X_train).squeeze()
  y_preds = torch.round(torch.sigmoid(y_logits))  # logits -> prediction probabilities -> prediction labels

  #computing the loss 
  loss = loss_fn(y_logits,y_train) # BECEwithlogitsloss takes logits as the first input
  #Calculate the Accuracy
  acc = accuracy_fn(y_true= y_train,
                    y_pred= y_preds)

  #Optimizer zero grad
  optimizer.zero_grad()

  #backpropogation
  loss.backward()

  #Optimizer step
  optimizer.step()

  #Training loop 
  model_3.eval()
  with torch.inference_mode():
    test_logits = model_3(X_test).squeeze()
    test_preds = torch.round(torch.sigmoid(test_logits))

    tets_loss = loss_fn(test_logits, y_test)
    test_acc = accuracy_fn(y_true= y_test,
                    y_pred= test_preds)


  if epoch %100 ==0:
    print(f"Epoch: {epoch} | Loss: {loss:.5f} , Acc: {acc:.2f}% | Test Loss: {test_loss:.5f} , Test Acc: {test_acc:.2f}%")
```

**Comment by ghost** [comment]
I'm still learning too so I'm not very confident in myself yet, but I'm pretty sure the code there is fine. You should check whether the optimizer has the right `params` of your `model_3` or model class implementation is okay, as well as previous steps. I recall having the same issue once and it was simply a matter of rerunning the model implementation cells and everything went smooth.

**Reply by GodRishUniverse** [reply]
I have re-run this multiple times and it is the same. I even checked the parameters entered but nothing is working - the model's graph shows very high accuracy but not these print statements

**Reply by ghost** [reply]
I think I got it. In the `torch.inference_mode()` context for testing you have a typo at `tets_loss = ...`. I think that might be it. What a silly thing right? I hope thats all there is to it :)

**Reply by GodRishUniverse** [reply]
@Yer-Marti I changed that too but still it shows the same error

**Reply by GodRishUniverse** [reply, marked as answer]
Nevermind, I resolved it. The problem was in the accuracy function - typo in `y_pred`:

Corrected version:
```
def accuracy_fn(y_true,y_pred):
   correct = torch.eq(y_true,y_pred).sum().item()
  acc = (correct/(len(y_pred)))*100
  return acc
```

Thanks for the help!

**Reply by ghost** [reply]
No problem, I'm glad you spotted it
